### DISCUSSION TITLE
Background color are not being rendered

### DISCUSSION THREAD

**Discussion by Angelfire** [tailwindlabs/tailwindcss#10574] [Help] [answered]
I have the following array:
```js
[
    {
        "hex": "#4A4F7D",
        "name": "Navy Blue",
        "pantone": "639 C",
        "rgb": "74, 79, 125"
    },
    {
        "hex": "#9CB7D2",
        "name": "Light Blue",
        "pantone": "631 C",
        "rgb": "156, 183, 210"
    },
    {
        "hex": "#FFD2E1",
        "name": "Lilac",
        "pantone": "677 C",
        "rgb": "255, 210, 225"
    },
    {
        "hex": "#FFF8EF",
        "name": "Ivory",
        "pantone": "12-5103 TCX",
        "rgb": "255, 248, 239"
    },
    {
        "hex": "#A9D7D9",
        "name": "Turquoise",
        "pantone": "7718 C",
        "rgb": "169, 215, 217"
    }
]
```

And I'm trying to do something like:
```js
export const Palette = ({ colors }: PaletteProps) => {
  return (
    <div className="mt-12 flex flex-col items-center">
      <div className="grid grid-cols-5 gap-3">
        {colors.map(color => (
          <div className={`bg-[${color.hex}] h-11 w-4`} key={color.name} />
        ))}
      </div>
    </div>
  )
}
  ```
  But, nothing happens, background colors are not being painted...

**Comment by andymccray** [comment]
[Per the docs](https://tailwindcss.com/docs/content-configuration#dynamic-class-names), Tailwind will `only find classes that exist as complete unbroken strings in your source files`. You’d need to use the full Tailwind class in your color array, e.g.

```js
{
  "hex": "bg-[#FFF8EF]",
  "name": "Ivory",
  "pantone": "12-5103 TCX",
  "rgb": "255, 248, 239"
}
```

**Reply by Angelfire** [reply]
The problem seems to be something else, I already tried it, I did a remap of my array and now I have objects of the type:
```js
{hex: 'bg-[#F8A06F]', name: 'Coral Reef', pantone: '17-1563', rgb: '248, 160, 111'}
```

But it still doesn't work...

**Reply by adamwathan** [reply]
I bet that file with the array isn’t included in your `content` configuration, if you add it there it will work.

**Reply by Angelfire** [reply]
The color array comes from a fetch I do to OpenAI, it's an experiment...

**Reply by adamwathan** [reply]
Yeah so it makes sense that this doesn't work when you think about it right? The stylesheet is compiled at build time, but the colors aren't known until the API call is made later at run-time, so it's not possible for those colors to end up in the stylesheet. Definitely best to use inline styles for this sort of thing 👍

**Comment by Angelfire** [comment, marked as answer]
It's funny that this way it works
```js
        {colors.map(color => (
          <div
            style={{ backgroundColor: `${color.hex}` }}
            className="h-24 w-9"
            key={color.name}
          />
        ))}
 ```

**Reply by adamwathan** [reply]
Yeah this works because inline styles are handled by the browser at run-time but Tailwind has to compile a stylesheet ahead of time at build time. For dynamic styles like this I would generally recommend using inline styles like you are here 👍🏻
